### PR TITLE
Validate objects based on openapi schema

### PR DESCRIPTION
**Pull request by lboeman**
Adds a class that uses the OpenAPI schema to validate objects on user input. Validation is performed using the Ajv library. The class also allows access to the underlying JSON of the spec for use printing descriptions etc.
As a proof of concept, this is implemented for tracking parameters.

### CODE REVIEW

File: dashboard/src/components/TrackingParameters.vue [hunk 1]
```diff
@@ -1,19 +1,34 @@
...
     </div>
     <div v-if="tracking == 'singleAxis'">
-      <b>Axis Tilt: </b><input v-model="parameters.axis_tilt" /><br />
-      <b>Axis Azimuth: </b><input v-model="parameters.axis_azimuth" /><br />
-      <b>Ground Coverage Ratio: </b><input v-model="parameters.gcr" /><br />
+      <b>Axis Tilt: </b><input v-model.number="parameters.axis_tilt" /><br />
+      <b>Axis Azimuth: </b
+      ><input v-model.number="parameters.axis_azimuth" /><br />
+      <b>Ground Coverage Ratio: </b
+      ><input v-model.number="parameters.gcr" /><br />
```
Comment (alorenzo175) [inline dashboard/src/components/TrackingParameters.vue:25]:
```suggestion
      <b>Axis Azimuth: </b>
      <input v-model.number="parameters.axis_azimuth" /><br />
      <b>Ground Coverage Ratio: </b>
      <input v-model.number="parameters.gcr" /><br />
```

File: dashboard/src/components/TrackingParameters.vue [hunk 2]
```diff
@@ -1,19 +1,34 @@
...
+        >{{ this.errors.tilt }}<br
+      /></span>
+      <b>Azimuth: </b
+      ><input type="number" v-model.number="parameters.azimuth" />
+      <span v-if="'azimuth' in this.definitions.properties">
+        {{ this.definitions.properties.azimuth.description }} </span
+      ><br />
+      <span style="color:#F00;" v-if="'azimuth' in this.errors"
+        >{{ this.errors.azimuth }}<br
+      /></span>
```
Comment (alorenzo175) [inline dashboard/src/components/TrackingParameters.vue:18]:
```suggestion
        {{ this.definitions.properties.tilt.description }} </span>
      <br />
      <span style="color:#F00;" v-if="'tilt' in this.errors">
        {{ this.errors.tilt }}<br />
      </span>
      <b>Azimuth: </b>
      <input type="number" v-model.number="parameters.azimuth" />
      <span v-if="'azimuth' in this.definitions.properties">
        {{ this.definitions.properties.azimuth.description }} </span>
      <br />
      <span style="color:#F00;" v-if="'azimuth' in this.errors">
        {{ this.errors.azimuth }}<br />
      </span>
```

**Review by alorenzo175 — APPROVED** [review]
your editor is doing some funky breaking in the html tags

**Comment by lboeman** [comment]
> your editor is doing some funky breaking in the html tags

That's actually the result of running `npm run lint --fix` I don't quite understand why it wants things like that, but eslint is very opinionated about it. I'll see if it's a rule we can turn off.

**Comment by lboeman** [comment]
I found the setting to avoid this issue, but it's going to create a mess for rebasing, so I'll correct it in the next PR.

**Pull request merged**